lib/messages/search-request.js

```diff
@@ -34,8 +34,15 @@ const isValidAttributeString = str => {
   if (/^@[a-zA-Z][\w\d.-]*$/.test(str) === true) {
     return true
   }
-  // ascii attribute names
-  if (/^[a-zA-Z][\w\d.;-]*$/.test(str) === true) {
+  /**
+  * ascii attribute names
+  *
+  * Validation inclusion of (=) and (*) characters supports the non-standard
+  * `range=<low>-<high>` ActiveDirectory extension as described in
+  * §3.1.1.3.1.3.3 (revision 57.0) of
+  * https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-adts/d2435927-0999-4c62-8c6d-13ba31a52e1a.
+  */
+  if (/^[a-zA-Z][\w\d.;=*-]*$/.test(str) === true) {
```

Comment:
What do you think about this?

```suggestion
  // Matches the non-standard `range=<low>-<high>` ActiveDirectory
  // extension as described in §3.1.1.3.1.3.3 (revision 57.0) of
  // https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-adts/d2435927-0999-4c62-8c6d-13ba31a52e1a.
  if (/^range=\d+-(\d+|\*)$/.test(str) === true {
    return true
  }
  
  // ascii attribute names per RFC 4512 §2.5
  if (/^[a-zA-Z][\w\d.;-]*$/.test(str) === true) {
```

Maybe it needs a bit of tweaking of other attributes are used in conjunction with the `range` attribute. But I think it would be clearer to make it its own test.

Reply:
When I run range as an attribute by itself using ldapsearch I get back no results. Same is true if I add multiple range extensions to a single attribute. I believe that is because it needs to be a unique extension per attribute. Also tried ldapsearch with multiple attributes, each with their own range extension, and that does return back results with the attributes matching the associated range values requested.

Would code to remove the range extension from the attribute string that we are testing look clear enough? In the example we are creating a new string to test out that supposedly only contains standard attribute characters, and would not mutate the original str. If that is OK, I would also like to add tests for invalid range options like `range=*-10` which should now fail.

```
   if (/^@[a-zA-Z][\w\d.-]*$/.test(str) === true) {
     return true
   }
+  // Remove non-standard `range=<low>-<high>` ActiveDirectory extension
+  // matches as described in §3.1.1.3.1.3.3 (revision 57.0) of
+  // https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-adts/d2435927-0999-4c62-8c6d-13ba31a52e1a.
+  // before testing for the standard ascii attribute names.
+  const standard = str.replace(/;range=\d+-(\d+|\*)/, "")
   // ascii attribute names
-  if (/^[a-zA-Z][\w\d.;-]*$/.test(str) === true) {
+  if (/^[a-zA-Z][\w\d.;-]*$/.test(standard) === true) {
     return true
   }
   return false
```